addons/web/static/src/js/views/view_dialogs.js

```diff
@@ -207,11 +222,32 @@ var FormViewDialog = ViewDialog.extend({
 
         return this;
     },
+    /**
+     * @override
+     * Adds a trash icon on header in full-screen mode
+     * if relational record is removable
+     */
+    set_buttons: function (buttons) {
+        var self = this;
+        this._super.apply(this, arguments);
+        var readonly = _.isNumber(self.options.res_id) && self.options.readonly;
+        if (!self.options.disable_multiple_selection && self.activeActions.delete && !readonly && self.isMobile) {
```

Comment:
Still not sure why all these conditions are needed, but you probably have an idea of why you need them here, right ?

Reply:
Yep all of these conditions are needed here :)

Reply:
What is the issue we will encounter if we remove the "disable_multiple_selection" condition ?

Reply:
This condition indicates that this is a x2m field, not m2o.(and we need remove / trash button only in x2m field. Its already used [Here](https://github.com/odoo/odoo/blob/984b87c378d0ca48b4f016e600316ea05e9c8eee/addons/web/static/src/js/views/view_dialogs.js#L105)  ).